autoPyTorch/pipeline/components/training/trainer/base_trainer.py

```diff
@@ -194,18 +192,17 @@ def prepare(
 
         # Weights for the loss function
         weights = None
-        kwargs = {}
-        if self.weighted_loss:
-            weights = self.get_class_weights(output_type, labels)
-            if output_type == BINARY:
-                kwargs['pos_weight'] = weights
-            else:
-                kwargs['weight'] = weights
-
-        criterion = criterion(**kwargs) if weights is not None else criterion()
+        kwargs: Dict[str, Any] = {}
+        # if self.weighted_loss:
```

Comment:
why have these lines been commented out?

Reply:
Because there was this issue with pos_weight and binary classifciation I talked to you about. Until its resolved I commented it out because it would throw errors on classification with two classes.

Reply:
I dont think we can treat binary classification as multi class classification because the traditional classifiers infer the metric to use and other things by checking the targets. Could you share the error that comes?

Reply:
The targets are not affected. What I suggested for binary classification was instead of predicting 1 output and using BCEWithLogitsLoss to predict 2 outputs and using standard CrossEntropyLoss, so we dont have to make this distinction with pos_weight and other stuff at all. What do you think?

Reply:
I think we are already predicting 2 outputs for binary classification see [here](https://github.com/automl/Auto-PyTorch/blob/a29dbee2b8f1b2520073a9ed1ce5d942c6ddefa6/autoPyTorch/datasets/base_dataset.py#L123). I think the confusion comes from the fact that I am checking if the task is binary and then making it pos_weight, but I think I should check if the criterion is BCELogitsWithLoss then we should use pos_weight, or remove this loss altogether. If you want to, could you change this in this PR? otherwise could you make an issue for this?

Reply:
I think the code you are linking to is exactly the change from me in this PR to avoid using binary cross entropy.

Reply:
yes, but even in the previous version [here](https://github.com/automl/Auto-PyTorch/blob/b48b952a2709f869f6e63511cfa0c94a889f3591/autoPyTorch/pipeline/components/setup/network_head/base_network_head.py#L41), its the same thing that is happening

Reply:
yep right